### PR TITLE
Remove add

### PR DESCRIPTION
**Pull request by ilyachur**
This PR should be merged first: https://github.com/openvinotoolkit/openvino/pull/1144

### CODE REVIEW

File: ngraph/test/backend_api.cpp [hunk 1]
```diff
@@ -58,35 +58,3 @@ TEST(backend_api, DISABLED_config_unsupported)
...
     EXPECT_FALSE(error == "");
 }
-
-#ifndef NGRAPH_JSON_DISABLE
-TEST(backend_api, save_load)
-{
-    Shape shape{2, 2};
-    auto A = make_shared<op::Parameter>(element::f32, shape);
-    auto B = make_shared<op::Parameter>(element::f32, shape);
-    auto f = make_shared<Function>(make_shared<op::Add>(A, B), ParameterVector{A, B});
```
Comment (tomdol) [inline ngraph/test/backend_api.cpp:68]: I think you should rather change the op in this test to a different op rather than removing it. It's not the addition operation itself itested here, it checks the save/load interface of a backend.

File: ngraph/test/provenance.cpp [hunk 2]
```diff
@@ -601,89 +601,3 @@ TEST(provenance, opset0_downgrade_pass_topk)
     };
     traverse_nodes({topk_v0}, tag_check, as_node_vector(topk_v1->input_values()));
 }
-
-TEST(provenance, opset1_upgrade_pass_graph)
```
Comment (tomdol) [inline ngraph/test/provenance.cpp:605]: The upgade pass is still used in the tests. It's supposed to be removed when we verify and confirm that the tests don't use `opset0` any more. I suggest to revert this change.

File: ngraph/test/runtime/interpreter/int_executable.cpp [hunk 3]
```diff
@@ -29,6 +29,7 @@
 #include "ngraph/pass/manager.hpp"
 #include "ngraph/serializer.hpp"
 #include "ngraph/util.hpp"
+#include "op/add.hpp"
```
Comment (tomdol) [inline ngraph/test/runtime/interpreter/int_executable.cpp:32]: I don't think this include is needed here.

File: ngraph/test/serialize.cpp [hunk 4]
```diff
@@ -144,21 +144,6 @@ TEST(serialize, friendly_name)
 }
 #endif
 
-TEST(serialize, existing_models)
```
Comment (tomdol) [inline ngraph/test/serialize.cpp:147]: This test has nothing to do with the Add operation. Why is it removed?

File: ngraph/test/serialize.cpp [hunk 5]
```diff
@@ -197,26 +182,6 @@ TEST(serialize, constant)
     EXPECT_TRUE(found);
 }
 
-TEST(benchmark, serialize)
```
Comment (tomdol) [inline ngraph/test/serialize.cpp:200]: Same here - why are you removing this test?

File: ngraph/test/backend_api.cpp (hunk 1 above)
Comment (ilyachur) [reply ngraph/test/backend_api.cpp:68]: Ok, I will change it. Thank you!

File: ngraph/test/provenance.cpp (hunk 2 above)
Comment (ilyachur) [reply ngraph/test/provenance.cpp:605]: I will recheck. Thank you!

File: ngraph/test/serialize.cpp (hunk 4 above)
Comment (ilyachur) [reply ngraph/test/serialize.cpp:147]: Models from this test cannot be serialised, because model contains v0::Add op.

File: ngraph/test/serialize.cpp (hunk 5 above)
Comment (ilyachur) [reply ngraph/test/serialize.cpp]: Model contains Add v0 operation
